components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.cc

```diff
@@ -327,6 +327,59 @@ void PolkadotSubstrateRpc::OnGetAccountBalance(
   return std::move(callback).Run(nullptr, WalletParsingErrorMessage());
 }
 
+void PolkadotSubstrateRpc::GetFinalizedHead(std::string_view chain_id,
+                                            GetFinalizedHeadCallback callback) {
+  auto url = GetNetworkURL(chain_id);
+
+  auto payload = base::WriteJson(
+      MakeRpcRequestJson("chain_getFinalizedHead", base::ListValue()));
+  CHECK(payload);
+
+  api_request_helper_.Request(
+      net::HttpRequestHeaders::kPostMethod, url, *payload, "application/json",
+      base::BindOnce(&PolkadotSubstrateRpc::OnGetFinalizedHead,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void PolkadotSubstrateRpc::OnGetFinalizedHead(GetFinalizedHeadCallback callback,
+                                              APIRequestResult api_result) {
+  if (!api_result.Is2XXResponseCode()) {
+    return std::move(callback).Run(std::nullopt, WalletInternalErrorMessage());
+  }
+
+  auto res = polkadot_substrate_rpc_responses::PolkadotFinalizedHead::FromValue(
+      api_result.value_body());
+
+  if (!res) {
+    return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+  }
+
+  if (res->error) {
+    if (res->error->message) {
+      return std::move(callback).Run(std::nullopt, res->error->message.value());
+    }
+    return std::move(callback).Run(std::nullopt, WalletInternalErrorMessage());
+  }
+
+  if (!res->result) {
+    return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+  }
+
+  std::string hash = std::move(*res->result);
+  if (!hash.starts_with("0x")) {
+    return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+  }
+
+  hash.erase(0, 2);
+  for (auto c : hash) {
+    if (!base::IsHexDigit(c)) {
+      return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+    }
+  }
+
```

Comment:
It would be preferred to return raw bytes instead of encoded in some way. Caller most likely would need to decode and validate hex again and we can avoid that.   

So let's do smth like 
```
std::vector<uint8_t> hash_bytes;
auto hash_hex = std::string_view(res->result.value());
if (!hash_hex.starts_with("0x") || !base::HexStringToBytes(hash_hex.substr(2u), &hash_bytes) {
  return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
}

return std::move(callback).Run(std::move(hash_bytes), std::nullopt);
```